streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -513,21 +515,15 @@ private void verifyGreaterOrEqual(final String[] keys,
             final Long value = keyValueStore.get(key);
             if (value != null) {
                 countState.put(key, value);
-            } else {
-                if (failIfKeyNotFound) {
-                    fail("Key not found " + key);
-                }
+            } else if (failIfKeyNotFound) {
+                fail("Key not found " + key);
             }
         }
 
         for (final Map.Entry<String, Long> actualWindowStateEntry : windowState.entrySet()) {
             if (expectedWindowedCount.containsKey(actualWindowStateEntry.getKey())) {
                 final Long expectedValue = expectedWindowedCount.get(actualWindowStateEntry.getKey());
                 assertTrue(actualWindowStateEntry.getValue() >= expectedValue);
-            } else {
```

Comment:
Why it is safe to remove this check and at line 540, after we have enlarged the window size?